### ISSUE TITLE
open map/planes in a new split instead of new tab

### ISSUE THREAD

**Issue by marttt** [q335r49/microviche#8]
Hi! What's the rationale behind opening Microviche in a new tab by default? Could you add an option for opening it in a new (horizontal) split instead?

I'm typically using a widescreen monitor set to portrait mode, and I'd like to use Microviche for storing "background" information for my main writing project. So I could have both things simultaneously in sight like so:

```
+---------------------------------+
|               main              |
|              project            |
|                                 |
|                                 |
+---------------------------------+
|  MV     | MV     | MV      |    |
|  plane1 | plane2 | plane3  | .. |
+---------------------------------+
```

Ideally, both the map and plane could be opened in this lower split. And there could also be an option for going fullscreen when a bigger overview of the map/planes is needed.

I find your plugin really interesting. Thanks for your work!

**Comment by q335r49** [comment]
Yeah, the lack of support for horizontal splits is an old problem that has been consistently deferred. The panning functions are all fairly hackish as there's no real API for determining window layout or adjusting multiple splits at once. Panning consists mostly of resizing a single window and making sure it worked by checking the size of another window, and the logic of this operation is reduced when one assumes a simple window layout.

So I've had this on the backburner for awhile but I keep on delaying it mostly because while it is certainly something that would make the script more robust it is also not something I myself use a great deal and so wouldn't really get the debugging attention it would need. It is planned for a future release though.

For now, have you considered writing a small script to toggle the main project window? Ie, to close the horizontal split when one needs to pan, and to have it open at other times.

Maybe like:

`nmap <f3> :if winheight(1)<&lines-&ch\|1 wincmd w\|q\|else\|topleft 25 split MainProject\|en<cr>`

Ie, "On pressing f3 in normal mode, if the height of the first window is less than the number of lines on the screen minus the command height, go to the first window and quit. Otherwise, split the window "MainProject" across the entire top of the screen and make it 25 lines tall."

or maybe something more sophisticated:

```
let MainWinHeight=25  
let MainWinName=''  
nno <silent> <f3> :exe winheight(1)<&lines-&ch? '1 wincmd w\|let MainWinHeight=winheight(1)\|let MainWinName=expand("%")\|q' : 'topleft '.MainWinHeight.'split '.fnameescape(MainWinName)<cr>
```

Here the script remembers and restores the previous height and file

**Comment by marttt** [comment]
Thanks for the explanation! I'll try the workaround, looks like it'll do
the job.

2014-05-22 23:21 GMT+03:00 q335r49 notifications@github.com:
